src/orquestra/quantum/circuits/_gates.py

```diff
@@ -434,6 +430,19 @@ def power(self, exponent: float) -> "Gate":
         return Power(self, exponent)
 
     def __str__(self):
+        non_commuting_gate_types = [ControlledGate, Power]
+        wrapped_gate = self.wrapped_gate
+        has_non_commuting_gate_type = False
+        while True:
```

Comment:
[suggestion] This is just a recursion that checks whether there's one of the wrapped gates is non-commuting, right?
We have duplication in another function, what about just extracting it to a separate function and then doing:
`if check_if_gate_has_non_commuting_component(wrapped_gate):` 
?

Reply:
Sure I'll do that.